### DISCUSSION TITLE
Ports tab list desynchronized from reality / validation wrong block

### DISCUSSION THREAD

**Discussion by jperelli** [community/community#5826] [Codespaces]
This occurs in the codespaces vscode (desktop) extension

From time to time the port tab shows the wrong ports information

steps to repro:
1. start some server and let vscode autodiscover a port
2. go to the ports tab and right click a port and select "make public"
3. the port dissapears from the list
4. before the port shows again in the list, right click in another port and "make public"
5. the ports information is now desynchronized from reality

I'd adventure to say that this might be a similar issue to what happens in reactjs/angularjs when I render an array and use the array index as idempotency key. In this case we should be using the port number as id.

In the image you can see that there is no 3000 port in the list, but at the same time the validation does not let me add a 3000 port.

![image](https://user-images.githubusercontent.com/1418446/134908442-e0bf4c11-3f33-475c-9f42-150abf0c6c1b.png)

In the next image you can see that the validation let's me add the same port that is already being shown in the list
![image](https://user-images.githubusercontent.com/1418446/134910384-c62e529f-9ac9-4002-95ae-966a46f0fed6.png)

I just found that a workaround to make the buggy 9222 port dissapear from the list is to stop the server in port 3000

**Comment by tanmayeekamath** [comment]
Hey @jperelli, thanks for reporting this with the detailed repro steps, we have filed an issue on our side. I'll share an update as soon as we fix this. Thanks for your patience.
